extensions/oidc/deployment/src/main/java/io/quarkus/oidc/deployment/OidcBuildStep.java

```diff
@@ -66,9 +69,15 @@ EnableAllSecurityServicesBuildItem security() {
 
     @Record(ExecutionTime.RUNTIME_INIT)
     @BuildStep(onlyIf = IsEnabled.class)
-    public void setup(OidcConfig config, OidcRecorder recorder, CoreVertxBuildItem vertxBuildItem,
-            BeanContainerBuildItem bc) {
-        recorder.setup(config, vertxBuildItem.getVertx(), bc.getValue());
+    public SyntheticBeanBuildItem setup(
+            OidcConfig config,
+            OidcRecorder recorder,
+            CoreVertxBuildItem vertxBuildItem) {
+        return SyntheticBeanBuildItem.configure(TenantConfigBean.class).unremovable().types(TenantConfigBean.class)
```

Comment:
Interesting, so this is how one can register the application scoped like singletons at runtime